samples/bluetooth/hap_ha/src/main.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2021-2022 Nordic Semiconductor ASA
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/capabilities.h>
+#include <zephyr/bluetooth/services/ias.h>
+
+#include "hap_ha.h"
+
+#define MANDATORY_SINK_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | \
+				BT_AUDIO_CONTEXT_TYPE_CONVERSATIONAL | \
+				BT_AUDIO_CONTEXT_TYPE_MEDIA | \
+				BT_AUDIO_CONTEXT_TYPE_LIVE)
+
+#define AVAILABLE_SINK_CONTEXT CONFIG_BT_PACS_SNK_CONTEXT
+#define AVAILABLE_SOURCE_CONTEXT CONFIG_BT_PACS_SRC_CONTEXT
+
+BUILD_ASSERT((CONFIG_BT_PACS_SNK_CONTEXT & MANDATORY_SINK_CONTEXT) == MANDATORY_SINK_CONTEXT,
+	     "Need to support mandatory Supported_Sink_Contexts");
+
+static uint8_t unicast_server_addata[] = {
+	BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL), /* ASCS UUID */
+	BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED, /* Target Announcement */
+	(((AVAILABLE_SINK_CONTEXT) >>  0) & 0xFF),
+	(((AVAILABLE_SINK_CONTEXT) >>  8) & 0xFF),
+	(((AVAILABLE_SOURCE_CONTEXT) >>  0) & 0xFF),
+	(((AVAILABLE_SOURCE_CONTEXT) >>  8) & 0xFF),
```

Comment:
Probably would nice to actually have some `BT_ENCODE_16`, `BT_ENCODE_24`, `BT_ENCODE_32`, etc. macros to handle these (similar to `BT_UUID_16_ENCODE` and `BT_UUID_32_ENCODE`), but that is outside the scope of this PR